bigquery_etl/cli/query.py

```diff
@@ -1412,12 +1419,19 @@ def initialize(ctx, name, sql_dir, project_id, dry_run, parallelism):
             init_files = Path(query_file.parent).rglob("init.sql")
 
             for init_file in init_files:
-                project = init_file.parent.parent.parent.name
                 client = bigquery.Client(project=project)
 
+                if skip_existing:
+                    try:
+                        client.get_table(full_table_id)
+                        # table exists; skip initialization
+                        continue
+                    except NotFound:
+                        # continue with creating the table
+                        pass
```

Comment:
It'd be nice if both init approaches were consistent with table existence and emptiness checks, so I'd suggest having a single set of checks before the init approach `if` statement (with a variable to store the table existence result for the `is_init()` approach to determine whether to deploy the table schema), rather than having the logic duplicated slightly differently like this.

Reply:
I moved the `skip_existing` logic to the beginning of the method. We do need to check if the provided files can be initialized, so if they are either `init.sql` files or have a `is_init()` statement. Only then we check via `client.get_table()` if the table exists, otherwise if we pass a wildcard `*` to `./bqetl initialize` we'd end up calling `client.get_table()` for all tables we have which takes forever to run.